src/ann_solo/reader.py

```diff
@@ -934,6 +937,85 @@ def read_query_file(filename: str) -> Iterator[MsmsSpectrum]:
     elif ext == '.mzxml':
         return read_mzxml(filename)
 
+
+def read_fasta(filename: str) -> Iterator[MsmsSpectrum]:
+    """
+    Read protein sequences from a FASTA file, and process them to predict
+    both genuine and decoy peptide spectra using Prosit.
+
+    Parameters
+    ----------
+    filename: str
+        The FASTA file name from which to read the protein sequences.
+
+    Returns
+    -------
+    Iterator[MsmsSpectrum]
+        An iterator of processed spectra.
+    """
+    missed_cleavages = 2
+
+    filename = filename  # FASTA file name
+    proteins = [
+        protein.sequence for protein in fasta.read(filename)
+    ]
+    tryptic_peptides = set().union(
+        *[
+            parser.cleave(peptide, "trypsin", missed_cleavages)
+            for peptide in proteins
+        ]
+    )
+    tryptic_peptides = list(tryptic_peptides)
+
+    ## Initializations of variables
+    tryptic_list_size = len(tryptic_peptides)
+    collision_energy = config.collision_energy
+
+    peptides = tryptic_peptides.copy()
+    peptides.extend(peptides)
+    precursor_charges = [2] * tryptic_list_size
```

Comment:
**todo:** Make the maximum and minimum precursor charge a user config.

Reply:
Fixed!